src/engine/strat_engine/lineardev.rs

```diff
@@ -0,0 +1,66 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use devicemapper::{DM, DevId, DeviceInfo, DmFlags};
+
+use engine::{EngineError, EngineResult, ErrorEnum};
+
+use super::blockdev::BlockDev;
+
+pub struct LinearDev {
+    name: String,
+    dev_info: Option<DeviceInfo>,
+}
+
+impl LinearDev {
+    pub fn new_lineardev(name: &str) -> LinearDev {
+        LinearDev {
+            name: name.to_owned(),
+            dev_info: None,
+        }
+    }
+
+    fn dm_table(&self, block_devs: &Vec<BlockDev>) -> Vec<(u64, u64, String, String)> {
+        let mut table = Vec::new();
+        let mut logical_start_sector = 0;
+        for block_dev in block_devs {
+            let (start, length) = block_dev.avail_range();
+            let dstr = block_dev.dstr();
+            let line = (logical_start_sector,
```

Comment:
Somewhere in the device mapper docs in the kernel it says that the very first logical start sector in a table must always be 0.

I'm not sure that this code would cause that to happen, although devicemapper-rs's ```load_table()``` might.